### ISSUE TITLE
Search result elements should be keyboard focusable

### ISSUE THREAD

**Issue by jakeisnt** [NixOS/nixos-search#258]
Currently, none of the search results can be navigated or selected with keyboard input alone, because they lack the HTML tags that tell the browser how to select them.

Adding the HTML attributes `tabindex` and `role` (button) to the heading of each search result would allow these headings to be expanded and contracted without the use of a mouse, which will enable more users to make better use of the site.

**Comment by samueldr** [comment]
In a more natural manner, using `<a>` (or `<button>`) element as part of one of the cells, like attribute name, should fix this without requiring `tabindex` (which can cause problems) or `role`.

Since they are elements that can be focused using the keyboard naturally in the browser, the browser already knows how to handle them. [It is how it was handled before the move to the new Elm app](https://github.com/NixOS/nixos-homepage/blob/b60ed4af1f3b94b9415312249a8725fbff6ab2aa/packages-explorer/src/gui/result.js#L33-L40).
